private/protocol/json/jsonutil/unmarshal.go

```diff
@@ -211,3 +227,33 @@ func unmarshalScalar(value reflect.Value, data interface{}, tag reflect.StructTa
 	}
 	return nil
 }
+
+func timeFromUnixString(d string) (time.Time, error) {
+	var secs, nsecs int64
+	if dot := strings.Index(d, "."); dot != -1 {
+		var err error
+		secs, err = strconv.ParseInt(d[:dot], 10, 64)
```

Comment:
If for some strange reason a service decides to write the time value as an exponent, which JSON allows, I think this would parse the incorrect value. e.g `1.23456e+10`.

Instead of processing the value as a string could we used something like `math.Modf` to extract the whole(seconds) and fractional components from the value?

Reply:
OK, I've now read https://www.cockroachlabs.com/blog/rounding-implementations-in-go/ which fills me with fear.

I think we should either use the function which will be math.Round in Go 1.10 (copied into the codebase), which requires a multiply/divide by 1e3 step, or https://github.com/montanaflynn/stats/blob/41c34e4914ec3c05d485e564d9028d8861d5d9ad/round.go#L5 which does "to 3 places" built-in.

Is it OK to vendor in the montanaflynn file? Or just copy that one file (it's MIT Licensed)?

Reply:
Thanks for the update @glasser. Using the Go.1.10 function would be the best option I think. The SDK can copy the function into the `/private/protocol` package with attributing the Go 1.10 stdlib as the source. It would be nice if the SDK used the actual stdlib implementation for Go 1.10 and higher. Using build flags is how the SDK generally does this for other version specific utilities.